sql/expression/function/math.go

```diff
@@ -105,4 +111,236 @@ func (r *Rand) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 	}
 
 	return rand.New(rand.NewSource(seed)).Float64(), nil
-}
\ No newline at end of file
+}
+
+var UintRegex, _ = regexp.Compile("^[1-9][0-9]*$*")
+var IntRegex, _ = regexp.Compile("^-?[1-9][0-9]*$")
+
+type UnaryMathFloatFuncLogic interface {
+	EvalFloat(float64) (interface{}, error)
+}
+
+type UnaryMathFuncLogic interface {
+	UnaryMathFloatFuncLogic
+	EvalUint(uint64) (interface{}, error)
+	EvalInt(int64) (interface{}, error)
+	EvalDecimal(decimal.Decimal) (interface{}, error)
+}
+
+type UnaryMathFloatFuncWrapper struct {
+	FlLogic UnaryMathFloatFuncLogic
+}
+
+func WrapUnaryMathFloatFuncLogic(logic UnaryMathFloatFuncLogic) UnaryMathFuncLogic {
+	return UnaryMathFloatFuncWrapper{logic}
+}
+
+func (wr UnaryMathFloatFuncWrapper) EvalFloat(n float64) (interface{}, error) {
+	return wr.FlLogic.EvalFloat(n)
+}
+
+func (wr UnaryMathFloatFuncWrapper) EvalUint(n uint64) (interface{}, error) {
+	return wr.FlLogic.EvalFloat(float64(n))
+}
+func (wr UnaryMathFloatFuncWrapper) EvalInt(n int64) (interface{}, error) {
+	return wr.FlLogic.EvalFloat(float64(n))
+}
+
+func (wr UnaryMathFloatFuncWrapper) EvalDecimal(dec decimal.Decimal) (interface{}, error) {
+	n, _ := dec.Float64()
+	return wr.FlLogic.EvalFloat(n)
+}
+
+type UnaryMathFunc struct {
+	expression.UnaryExpression
+	Name  string
+	Logic UnaryMathFuncLogic
+}
+
+func NewUnaryMathFunc(name string, logic UnaryMathFuncLogic) func(e sql.Expression) sql.Expression {
+	return func(e sql.Expression) sql.Expression {
+		return &UnaryMathFunc{expression.UnaryExpression{Child: e}, name, logic}
+	}
+}
+
+// Eval implements the Expression interface.
+func (mf *UnaryMathFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	val, err := mf.Child.Eval(ctx, row)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if val == nil {
+		return nil, nil
+	}
+
+	// Fucking Golang
```

Comment:
And then they all boil down to float64(x) anyway

Reply:
What do you mean I cast to float64 anyway?

Reply:
Same comment as below basically